notes/archipelago

```diff
@@ -60,3 +60,86 @@ potentially be very efficient about defragging early on?
+    algorithm for randomizing entrances. This algorithm supports grouping of entrances, which
+	means we'll be able to use it for both house and area shuffle. (Goa shuffle may still
+	need special handling, due to invalid combinations of floors.)
+  - Easier to read and maintain. The approach which wrapped all of Crystalis Randomizer into 
+    a single file and then called into it from Python code resulted in a file over 100K lines 
+	long. Getting that reviewed in a code review when eventually PR'ing to AP core (should that 
+	happen) seems highly improbable.
+
+Cons:
+  - Error prone. While I'm reasonably confident in the implementation of the logic, it's
+    certainly more likely to fail than reading the logic directly from the randomizer itself.
+  - Potentially results in divergent codebases. Again, while I'm trying to maintain parity
+    with this codebase, any logic change in the base randomizer now necessitates a mirrored
+	change in the .apworld. This can be particularly problematic if this .apworld is accepted
+	by the AP Core devs, as that ties .apworld changes down to AP's release schedule.
+  - Extra website step is not ideal UX for the player. It adds an extra roadblock to playing,
+    and works differently to most other ROM games. (That said, there is precedent in the Final
+	Fantasy Mystic Quest implementation.
+  - Some features may be too complex to represent in the .apcrys file. In particular, I'm not
+    currently sure how to implement Wm - randomize maps - in a way that the logic can be read 
+	reasonably and a representation can be sent to this codebase to rearrange the map in the
+	way logic expects.
+
+In my opinion, the pros overall outweigh the cons, hence why I settled on this approach, but
+I could certainly see another developer taking a different approach.
+
+Feature To-Do list:
+  -logic and implementation of flags not yet implemented:	
+    -Wa, Wh (using GER, not in initial release)
+    -Wm (not a clue; implementation TBD, not in initial release)
+  -Archipelago specific features:
+    -Unit tests
+    -Deathlink >:)
+	-Gifting interface? (maybe take over pawn-shops? not convinced this makes sense, but I'm willing to consider it)
+	-Option for local items being baked into the ROM? (Probably not, given how item flags end up working, but you never know)
+	
+Other notes:
+  Currently we don't bake any information into the ROM about what items are. However, this information
+  is available from the Archipelago generation. I am resistant to putting it into the patch data, as it
+  would result in the patch data doubling as a personal spoiler log for anyone with a little technical 
+  savvy. The might be a world where only local items get baked into the ROM, in which case I can just use
+  item IDs, which would do a sufficient amount of obfuscation, hopefully. I've also toyed around with the
+  idea of making the Crystalis Randomizer code connect to the Archipelago server as a client; then all the
```

Comment:
Likely impossible with our current JS-only hosting situation, due to CSP or other sorts of security restrictions.  My guess is we'd require at least some sort of relay server to do the communication.

Reply:
That's fair.  Honestly it's kind of a janky idea to begin with, so probably not worth exploring in depth at this time.